benchmark_v2/sort_cases_benchmark.py

```diff
@@ -1,256 +1,289 @@
+
+
+def get_nbytes(data):
+    if isinstance(data, (ak.pdarray, ak.Strings)):
+        return data.nbytes
+    else:
+        return sum(x.nbytes for x in data)
+
+
+def do_argsort(data, algo):
+    if isinstance(data, (ak.pdarray, ak.Strings)):
+        return ak.argsort(data, algo)
+    else:
+        return ak.coargsort(data, algo)
+
+
+def run_sort(benchmark, name, data, sort_fn):
+    # NumPy fallback
+    if pytest.numpy:
+        if isinstance(data, tuple):
+            data_np = tuple(x.to_ndarray() for x in data)
+
+            def sort_op():
+                np.lexsort(data_np[::-1])
+                return sum(x.nbytes for x in data_np)
+        else:
+            data_np = data.to_ndarray()
+
+            def sort_op():
+                np.argsort(data_np)
+                return data_np.nbytes
+
+        backend = "NumPy"
+
+    else:
+
+        def sort_op():
+            p = sort_fn(data)
+            if pytest.correctness_only:
+                if isinstance(data, tuple):
+                    sorted_data = [d[p] for d in data]  # noqa: F841
+                    # Assume lexsorted comparison is hard to check; skip for now
+                else:
+                    assert ak.is_sorted(data[p])
+            return get_nbytes(data)
+
+        backend = "Arkouda"
+
+    numBytes = benchmark.pedantic(sort_op, rounds=pytest.trials)
+
+    benchmark.extra_info["description"] = f"{name} via {backend}"
+    benchmark.extra_info["problem_size"] = data[0].size if isinstance(data, tuple) else data.size
+    benchmark.extra_info["backend"] = backend
+    benchmark.extra_info["transfer_rate"] = "{:.4f} GiB/sec".format(
+        (numBytes / benchmark.stats["mean"]) / 2**30
+    )
+
+
+@pytest.mark.benchmark(group="AK_Sort_Cases")
+@pytest.mark.parametrize("algo", SortingAlgorithm)
+@pytest.mark.parametrize("dtype", TYPES)
+@pytest.mark.parametrize("bits", ("16-bit", "32-bit", "64-bit"))
+def bench_random_uniform(benchmark, algo, dtype, bits):
+    """
+    Uniformly distributed integers of 1, 2, and 4 digits.
+    Uniformly distributed reals in (0, 1)
+    """
+    if dtype in pytest.dtype:
+        N = pytest.prob_size
+
+        if dtype == "int64":
+            if bits == "16-bit":
+                data = ak.randint(0, 2**16, N)
+            elif bits == "32-bit":
+                data = ak.randint(0, 2**32, N)
+            else:
+                data = ak.randint(-(2**63), 2**63, N)
+
+            run_sort(benchmark, "random_uniform", data, lambda x: do_argsort(x, algo))
+            nbytes = get_nbytes(data)
+            benchmark.extra_info["description"] = "Measures the performance of random_uniform sort case"
+            benchmark.extra_info["problem_size"] = pytest.prob_size
+            benchmark.extra_info["transfer_rate"] = "{:.4f} GiB/sec".format(
+                (nbytes / benchmark.stats["mean"]) / 2**30
+            )
+        elif bits == "64-bit":  # float64 - We only want this to run once per algorithm, not 3 times
```

Comment:
There are assignments to `benchmark.extra_info[...]` both in `run_sort` and outside/after `run_sort`. Only one set should remain.

Also `nbytes = get_nbytes(data` after `run_sort` is called may not be needed.

The same applies to all invocations of `run_sort` in this file.